### PR TITLE
Pending updates for team and tash dashboard

### PR DESCRIPTION
**Pull request by CodeFreak2002**
I will merge this commit after resolving the conflicts. Current deployment is from main, so some functionalities will not work. After this merge I will redeploy the server.

### CODE REVIEW

File: server/models/team.js [hunk 1]
```diff
@@ -25,11 +25,21 @@ const teamSchema = Schema({
     },
 
     members : {
-        type : [{ type : Schema.Types.ObjectId, ref : 'student'}]
+        type : [{ type : Schema.Types.ObjectId, ref : 'student' }]
```
Comment (Ashwin-1709) [inline server/models/team.js:28]: remove extra space.

File: server/routes/task.js [hunk 2]
```diff
@@ -4,17 +4,25 @@ const Team = require('../models/team')
...
         return res.status(403).send("Only team manager can create tasks!")
 
     let task = new Task({
+        title : req.body.title.toString(),   
         description : req.body.description.toString(),
         deadline: req.body.deadline ? req.body.deadline.toString() : null,
-        completionStatus : "not started",
-        team : req.body.team.toString()
+        completionStatus : "Available",
+        team : req.body.team.toString(),
```
Comment (Ashwin-1709) [inline server/routes/task.js:25]: remove comma

File: server/routes/task.js [hunk 3]
```diff
@@ -24,33 +32,98 @@ router.post("/create/" , async(req , res) => {
...
-    return res.status(200).send("Task completed");
+    
+    await Task.findByIdAndUpdate(
+
+        req.body.task,
+        
+        { endTime : curTime, completionStatus : "In Review" },
+        
+        async function (err, task)
+        {
```
Comment (Ashwin-1709) [inline server/routes/task.js:81]: minor: braces style can be made consistent throughout the project. some places use
`function() {

}`

while some use 
`function()
{

}`

File: server/routes/task.js [hunk 4]
```diff
@@ -24,33 +32,98 @@ router.post("/create/" , async(req , res) => {
...
+            if (err)
+                console.log(err)
+            else
+            {
+                console.log(task.team)
+                await Team.findByIdAndUpdate(
+                    task.team,
+                    { $push: {review: req.body.task} }
```
Comment (Ashwin-1709) [inline server/routes/task.js:89]: minor: braces style

File: server/routes/task.js [hunk 5]
```diff
@@ -24,33 +32,98 @@ router.post("/create/" , async(req , res) => {
...
+        await Task.findByIdAndUpdate(
+            req.body.task,
+            { completionStatus : "Completed" }
+        ).clone()
+    }
+    else
+    {
+        await Task.findByIdAndUpdate(
+            req.body.task,
+            { completionStatus : "Ongoing" }
```
Comment (Ashwin-1709) [inline server/routes/task.js:116]: minor: braces style

File: server/routes/task.js [hunk 6]
```diff
@@ -24,33 +32,98 @@ router.post("/create/" , async(req , res) => {
...
+        ).clone()
+ 
+    }
+
+    await Team.findByIdAndUpdate(
+        task.team,
+        { $pull: {review: req.body.task} }
```
Comment (Ashwin-1709) [inline server/routes/task.js:123]: minor: braces style

File: server/routes/task.js [hunk 7]
```diff
@@ -24,33 +32,98 @@ router.post("/create/" , async(req , res) => {
...
+    ).clone();
+    
+    return res.status(200).send("Task sent for review!");
+});
+
+router.post("/accept", async(req, res) => {
+
+    let task = await Task.findById(req.body.task).clone()
+
+    if (!task)
```
Comment (Ashwin-1709) [inline server/routes/task.js:102]: minor: Null check can be done use  isNull function from the Underscore library for a more reliable checking.

example:
const myVar = null;
console.log(_.isNull(myVar)); // true

**Review by Ashwin-1709 — APPROVED** [review]
Minor comments added.

File: server/routes/task.js (hunk 3 above)
Comment (CodeFreak2002) [reply server/routes/task.js:81]: Linting issues can be fixed globally using a linter which can be added to a pre-commit hook. To be taken up in a future commit.

File: server/routes/task.js (hunk 7 above)
Comment (CodeFreak2002) [reply server/routes/task.js:102]: merging for now, will resolve in a later commit

File: server/routes/task.js (hunk 2 above)
Comment (CodeFreak2002) [reply server/routes/task.js]: done

**Pull request merged**